lib/src/query.c

```diff
@@ -2285,6 +2311,15 @@ bool ts_query_cursor_did_exceed_match_limit(const TSQueryCursor *self) {
   return self->did_exceed_match_limit;
 }
 
+uint32_t ts_query_cursor_match_limit(const TSQueryCursor *self) {
+  return self->capture_list_pool.max_capture_list_count;
+}
+
+void ts_query_cursor_set_match_limit(TSQueryCursor *self, uint32_t limit) {
+  assert(limit > 0);
```

Comment:
I'm kind of inclined to just allow zeros here. Even though it would cause a behavior that's not very useful (basically no matches allowed), I don't think it entails an unrecoverable invariant violation that merits a hard abort.

Reply:
That should work — I had put it in because it seemed daft, not because I thought it would break anything.